### PR TITLE
[MOL-16390][Tung] Update touch area of accordion

### PR DESCRIPTION
**Pull request by QuocTung95**
**Changes**
move onClick={handleExpandCollapseClick} from ExpandCollapseButton to TitleContainer
create css cursor: pointer for TitleContainer

- delete branch

- You may refer to this [[MOL-16390](https://jira.ship.gov.sg/browse/MOL-16390)]
- **Demo**

https://github.com/user-attachments/assets/c439bc0d-b1c9-433f-960b-18f768f54e30

### CODE REVIEW

File: src/accordion/accordion-item.tsx
Comment (qroll) [inline src/accordion/accordion-item.tsx:1]: this needs to account for the `collapsible` prop

if `collapsible` is false, then the title shouldn't be interactive, and the cursor should remain as the default

Comment (qroll) [inline src/accordion/accordion-item.tsx:1]: can I confirm if this behaviour should apply to all accordions? or only in specific scenarios?

if the latter, then it should be configurable through a prop

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/accordion/accordion-item.tsx
Comment (QuocTung95) [reply src/accordion/accordion-item.tsx:1]: it should be apply for all, confirmed form UX

Comment (QuocTung95) [reply src/accordion/accordion-item.tsx:1]: But this morning QE said "now the scope has changed. I would like to clarify this with UX on exact accordions that requires this change" in the chat. So wait for me uppdate

Comment (QuocTung95) [reply src/accordion/accordion-item.tsx:1]: With new this Accordion we make it as default ( touch area whole title - confirmed ). And with BoxContainer we keep it as old version

Comment (QuocTung95) [reply src/accordion/accordion-item.tsx:1]: checked collapsible

**Review by qroll — APPROVED** [review]

**Pull request merged**